tests/test_fill.py

```diff
@@ -0,0 +1,78 @@
+import dpctl
+import numpy as np
+import pytest
+from dpctl.utils import ExecutionPlacementError
+from numpy.testing import assert_array_equal
+
+import dpnp as dnp
+
+
+def test_fill_non_scalar():
+    a = dnp.ones(5, dtype="i4")
+    val = dnp.ones(2, dtype="i4")
+
+    with pytest.raises(ValueError):
+        a.fill(val)
+
+    with pytest.raises(TypeError):
+        a.fill(dict())
+
+
```

Comment:
and also it'd meaningful to cover bool fill value:
```suggestion

def test_fill_bool():
    a = dnp.full(5, fill_value=7, dtype="i4")

    a.fill(True)
    assert_array_equal(a, 1)

```